packages/grafana-ui/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,82 @@
+import { css } from '@emotion/css';
+import { FocusScope } from '@react-aria/focus';
+import React, { useState } from 'react';
+import { usePopperTooltip } from 'react-popper-tooltip';
+import { CSSTransition } from 'react-transition-group';
+
+import { ReactUtils } from '../../utils';
+import { Portal } from '../Portal/Portal';
+import { TooltipPlacement } from '../Tooltip/types';
+
+export interface Props {
+  overlay: React.ReactElement | (() => React.ReactElement);
+  placement?: TooltipPlacement;
+  children: React.ReactElement;
+  /** Defaults to click */
+  trigger?: Array<'click' | 'hover'>;
+}
+
+export const Dropdown = React.memo(({ children, overlay, placement, trigger = ['click'] }: Props) => {
+  const [show, setShow] = useState(false);
+  const { getArrowProps, getTooltipProps, setTooltipRef, setTriggerRef, visible } = usePopperTooltip({
+    visible: show,
+    placement: placement,
+    onVisibleChange: setShow,
+    interactive: true,
+    delayHide: 200,
+    delayShow: 150,
+    offset: [0, 8],
+    trigger: trigger,
+  });
+
+  const animationDuration = 150;
+  const animationStyles = getStyles(animationDuration);
+
+  const onOverlayClicked = () => {
+    setShow(false);
+  };
+
+  return (
+    <>
+      {React.cloneElement(children, {
+        ref: setTriggerRef,
+      })}
+      {visible && (
+        <Portal>
+          <FocusScope autoFocus>
```

Comment:
🤔 we could use instead `restoreFocus` , considering this component can be used for different scenarios we are not really sure we want to have it as the first focusable element on the page. Also with restore focus we make sure to put the focus back when the user select the option in the dropdown :)

Reply:
restoreFocus has some bad behaviors in that it focuses the last focus element causing tooltips to show and it's not really needed here. 

> considering this component can be used for different scenarios we are not really sure we want to have it as the first focusable element on the page

? Not sure I undestand. This is inside the Portal and only active when the menu is open,  The autoFocus is needed here to move focus to the newly opened menu

Reply:
I missed the <Portal> line🤦🏾‍♀️, in this case, you are right, we can use autoFocus 👌🏾

Reply:
I don't think it's Portal that is the important bit, the menu/overlay is not added to DOM until it's visible :)

Reply:
Aha! Thank you, I think I get the logic and found the mistake I was making, for a weird reason I was associating this part with the `triggerRef` children instead of the Dropdown.

Reply:
@axelavargas removed the trigger property to only support click right now as support hover+focus was tricky together with a menu that has autoFocus. 

Maybe we can add hover+focus support later when we have a clear need for it